code/modules/holodeck/computer.dm

```diff
@@ -261,12 +261,7 @@ GLOBAL_LIST_INIT(typecache_holodeck_linked_floorcheck_ok, typecacheof(list(/turf
 	//makes sure that any time a holoturf is inside a baseturf list (e.g. if someone put a wall over it) its set to the OFFLINE turf
 	//so that you cant bring turfs from previous programs into other ones (like putting the plasma burn turf into lounge for example)
 	for(var/turf/closed/holo_turf in linked)
-		for(var/baseturf in holo_turf.baseturfs)
-			if(ispath(baseturf, /turf/open/floor/holofloor))
-				var/list/copy = holo_turf.baseturfs.Copy()
-				copy -= baseturf
-				copy += /turf/open/floor/holofloor/plating
-				holo_turf.baseturfs = baseturfs_string_list(copy, holo_turf)
+		holo_turf.replace_baseturf(/turf/open/floor/holofloor, /turf/open/floor/holofloor/plating)
```

Comment:
Different behavior, @Kylerace said this is what it was meant to be in the first place.